hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/OzoneManagerServiceProviderImpl.java

```diff
@@ -623,6 +639,34 @@ public boolean syncDataFromOM() {
     return true;
   }
 
+  private void printFileAndKeyTableCount() throws IOException {
+    Table fileTable = omMetadataManager.getTable("fileTable");
+    Table keyTable = omMetadataManager.getTable("keyTable");
+    if (keyTable == null) {
+      LOG.error("Table keyTable not found in OM Metadata.");
+    }
+
+    if (LOG.isDebugEnabled() && null != keyTable) {
+      try (TableIterator<String, ? extends Table.KeyValue<String, ?>> iterator
+               = keyTable.iterator()) {
+        long count = Iterators.size(iterator);
+        LOG.debug("keyTable Table count: {}", count);
```

Comment:
We can combine null checks and debug log conditions to avoid repetitive code, and also write it in such a way that we can add any other table in it in the future if necessary.

```
private void printFileAndKeyTableCount() throws IOException {
    printTableCount("fileTable");
    printTableCount("keyTable");
}

private void printTableCount(String tableName) throws IOException {
    Table table = omMetadataManager.getTable(tableName);
    if (table == null) {
        LOG.error("Table {} not found in OM Metadata.", tableName);
        return;
    }
    if (LOG.isDebugEnabled()) {
        try (TableIterator<String, ? extends Table.KeyValue<String, ?>> iterator = table.iterator()) {
            long count = Iterators.size(iterator);
            LOG.debug("{} Table count: {}", tableName, count);
        }
    }
}
```

Reply:
Ok sure. will make it generic.

Reply:
Ok sure. will make it generic.